File: src/com/dafttech/terra/game/world/items/ItemSaplingArrow.java
package com.dafttech.terra.game.world.items;

import com.badlogic.gdx.Gdx;
import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.entities.EntitySaplingArrow;
import com.dafttech.terra.game.world.entities.Player;

public class ItemSaplingArrow extends ItemArrow {
    @Override
    public boolean spawnEntity(Player causer, Vector2 position) {
        EntitySaplingArrow a = new EntitySaplingArrow(causer.getPosition(), causer.worldObj);
        a.setVelocity(Vector2.getMouse().sub(new Vector2(Gdx.graphics.getWidth() / 2, Gdx.graphics.getHeight() / 2)).mul(0.2f));
        return true;
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/items/ItemSaplingArrow.scala
package com.dafttech.terra.game.world.items

import com.badlogic.gdx.Gdx
import com.dafttech.terra.engine.input.MousePos
import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.game.world.entities.EntitySaplingArrow
import com.dafttech.terra.game.world.entities.models.EntityLiving

class ItemSaplingArrow extends ItemArrow {
  override def spawnEntity(causer: EntityLiving, position: Vector2d): Boolean = {
    val a = new EntitySaplingArrow(causer.getPosition)(causer.world)

    val velocity = (MousePos.vector2d - (Gdx.graphics.getWidth / 2, Gdx.graphics.getHeight / 2)) * 0.2
    a.body.setLinearVelocity(velocity.x.toFloat, velocity.y.toFloat)
    true
  }
}